### PR TITLE
First draft

### PR DESCRIPTION
**Pull request by krl**
I would like it to actually do init automatically as well, then i'll update the readme.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+function configureNode (node, conf, cb) {
+  var delay = 0
+  var promises = _.map(conf, function (value, key) {
+    var def = Q.defer()
+    setTimeout(function () {
+      run('ipfs', ['config', key, value])
+        .on('error', function (err) { cb(err) })
+        .on('end', function () { def.resolve(true) })
+    }, delay * 200)
+    delay++
```
Comment (jbenet) [inline index.js]: may want to say why it has to run this way?

File: package.json [hunk 2]
```diff
@@ -4,19 +4,32 @@
   "description": "simple controls for an ipfs node",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "mocha test"
```
Comment (travisperson) [inline package.json]: Missing comma.

File: package.json [hunk 3]
```diff
@@ -4,19 +4,32 @@
...
+    "lint": "git diff --name-only --cached --relative | egrep .js$ | xargs --no-run-if-empty standard"
   },
+  "pre-commit": [
+    "test"
+  ]
```
Comment (travisperson) [inline package.json]: Missing comma.

**Comment by krl** [comment]
test

**Comment by jbenet** [comment]
nice screenshot

**Comment by jbenet** [comment]
@krl this is great

File: index.js (hunk 1 above)
Comment (krl) [inline index.js]: It seems there can be a race condition when you're setting two config variables at the same time. I will make an issue of this in my next sitting.

Comment (travisperson) [inline index.js]: Sounds like a good use case for waterfall promises?

File: test/test.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
+var run = require('comandante')
+var ipfsd = require('../index.js')
+var assert = require('assert')
+var fs = require('fs')
+var rimraf = require('rimraf')
+
+// happy lint
+// var it, before, after
```
Comment (travisperson) [inline test/test.js]: Does this actually work to allow standard to not complain about undefined? I usually see something like:
```
/* global beforeEach, describe, it */
```

File: index.js (hunk 1 above)
Comment (travisperson) [inline index.js]: See: https://www.npmjs.com/package/promise-waterfall

**Comment by krl** [comment]
@travisperson thank you for pointing me at waterfall promises, will def be using this from now on!

Pushed new version with this method

**Comment by bcomnes** [comment]
:+1: Thanks for doing this!

**Comment by bcomnes** [comment]
What is the relationship between https://www.npmjs.com/package/ipfs-api and https://www.npmjs.com/package/go-ipfs?

**Comment by jbenet** [comment]
@bcomnes ipfs-api is a js lib for consuming/using the ipfs api provided by go-ipfs. (it's an http server run by go-ipfs, and other implementations should probably support it (match the api) too)

**Comment by jbenet** [comment]
This LGTM

**Comment by jbenet** [comment]
@krl shall we merge this?

**Comment by bcomnes** [comment]
Baller! Nice work @krl   Just had a chance to play around with it.  Seems to work as expected

**Pull request merged**